packages/sdk/src/cosmwasmclient.ts

```diff
@@ -0,0 +1,228 @@
+import { Encoding } from "@iov/encoding";
+
+import { makeSignBytes, marshalTx } from "./encoding";
+import { findAttribute, Log, parseLogs } from "./logs";
+import { RestClient } from "./restclient";
+import {
+  Coin,
+  MsgExecuteContract,
+  MsgInstantiateContract,
+  MsgStoreCode,
+  StdFee,
+  StdSignature,
+} from "./types";
+
+const defaultUploadFee: StdFee = {
+  amount: [
+    {
+      amount: "5000",
+      denom: "ucosm",
+    },
+  ],
+  gas: "1000000", // one million
+};
+
+const defaultInitFee: StdFee = {
+  amount: [
+    {
+      amount: "5000",
+      denom: "ucosm",
+    },
+  ],
+  gas: "500000", // 500k
+};
+
+const defaultExecFee: StdFee = {
+  amount: [
+    {
+      amount: "5000",
+      denom: "ucosm",
+    },
+  ],
+  gas: "200000", // 200k
+};
+
+export interface SigningCallback {
+  (signBytes: Uint8Array): Promise<StdSignature>;
+}
+
+interface SigningData {
+  readonly senderAddress: string;
+  readonly signCallback: SigningCallback;
+}
+
+export interface GetNonceResult {
+  readonly accountNumber: number;
+  readonly sequence: number;
+}
+
+export interface PostTxResult {
+  readonly logs: readonly Log[];
+  readonly rawLog: string;
+  readonly transactionHash: string;
+}
+
+export class CosmWasmClient {
+  public static makeReadOnly(url: string): CosmWasmClient {
```

Comment:
I like this separation of readable and writable.

I assume the readable one has methods that would throw errors rather than a different interface.
Good for first version, maybe we can make `clas CosmWasmSigningClient extends CosmWasmClient` for write vs read.

Reply:
"read only" might be misleading as it actually means _no signing support_. You can use a readonly client to post transactions signed elsewhere (which is what we do in the BCP implementation).